### ISSUE TITLE
Unexpected token � in JSON at position 0

### ISSUE THREAD

**Issue by SukantGujar** [th0r/webpack-bundle-analyzer#47]
#### Issue description
wba is reporting this error on a valid stats.json file.

#### Technical info
* Webpack Bundle Analyzer version: 2.2.3
* Webpack version: 2.2.1
* Node.js version: 6.9.1
* npm/yarn version: 3.10.8/0.19.1
* OS: Windows 10

#### Debug info
How do you use this module? As CLI utility or as plugin?
As CLI
Generated stats with 
`webpack --profile --json > dist/stats.json`

If CLI, what command was used? `webpack-bundle-analyzer dist/stats.json dist`

What other Webpack plugins were used?
```
LodashModuleReplacementPlugin
webpack.optimize.CommonsChunkPlugin
```
It would be nice to also attach webpack stats file. Attached!
[stats.zip](https://github.com/th0r/webpack-bundle-analyzer/files/769101/stats.zip)

**Comment by valscion** [comment]
For some weird reason, your `stats.json` has a [Byte Order Mark (BOM)](https://en.wikipedia.org/wiki/Byte_order_mark) in there, which causes the reporting to fail.

I don't know how that got in there... [This issue in webpack repository](https://github.com/webpack/webpack/issues/2868) seems to be the only one mentioning a BOM character.

Are you sure you haven't opened the `stats.json` in some text editor before trying to use `webpack-bundle-analyzer` CLI?

**Comment by SukantGujar** [comment]
That's odd. But no I haven't opened it up _before_, only after the error was displayed, I used Sublime to view it. And the one attached in this thread is straight from the tool. A possibility is that the LodashModuleReplacementPlugin plugin is somehow causing it, I had an issue in the past where a plugin injected its log into the webpack output, thereby breaking the json.

**Comment by valscion** [comment]
I don't think this is a bug in webpack-bundle-analyzer as the BOM shouldn't get into the stats file in the first place.

The bug is in whatever put the BOM in your stats.json, not WBA

**Comment by th0r** [comment]
This `stats.json` file is broken. Even `node` can't require it:
```
> var stats = require("./samples/stats.1.json");
SyntaxError: /webpack-bundle-analyzer/samples/stats.1.json: Unexpected token � in JSON at position 0
```

**Comment by SukantGujar** [comment]
Thanks @valscion and @th0r. After saving the generated file as UTF-8 in Sublime, the stats file displayed correctly with wba. FWIW, the source code is [available here](https://github.com/SukantGujar/webpack-split-experiments), in case you wish to try it out at your end. I tested it without LodashReplacementPlugin, but the issue remained, so I guess it could either be a weird environmental issue, or something related with webpack.

**Comment by SukantGujar** [comment]
Turned out to be an [encoding issue with Powershell](https://www.kongsli.net/2012/04/20/powershell-gotchas-redirect-to-file-encodes-in-unicode/). For future users stumbling across this, use the following command to get build stats file in Powershell -
`webpack --profile --json | Out-file 'dist/stats.json' -Encoding OEM`

**Comment by valscion** [comment]
Hey that's a great find, @SukantGujar! I created a PR to document this in case someone else stumbles upon the same issue: #51

**Comment by NullVoxPopuli** [comment]
I have this issue on linux. There is no out-file command... so.. yeah

**Comment by valscion** [comment]
Huh. This:

```
iconv: illegal input sequence at position 1791728
```

looks like your stats.json has something else than a BOM in the beginning.

Could you create a new issue @NullVoxPopuli with your case and fill in the issue template?

**Comment by BrianMikinski** [comment]
> Turned out to be an [encoding issue with Powershell](https://www.kongsli.net/2012/04/20/powershell-gotchas-redirect-to-file-encodes-in-unicode/). VSCode used it for integrated terminal on my machine. For future users stumbling across this, use the following command to get build stats file in Powershell -
> `webpack --profile --json | Out-file 'dist/stats.json' -Encoding OEM`

I know this issue is closed but something to note is that "Out-file" is a powershell command and will not work if you are running it within an npm script such as 

``` console
npm run scriptName
```

You need to the windows command from an actual command prompt

**Comment by valscion** [comment]
What would that look like in practice? Is there a way to amend the documentation to cover this case @BrianMikinski ?

**Comment by BrianMikinski** [comment]
First off @valscion, this plugin is awesome, great work. I just discovered it a couple of days ago and it's fantastic.

As for my issue running the analyzer, I've figured out what was happening and a good path forward. Initially, when trying to get bundle analyzer up and running, I was using this npm run script to generate the stats.json file - 

```javascript
"buildStats": "webpack --config=webpack.config.js --env dev --profile --json > stats.json",
```
and when I went to run webpack I got the following error

![image](https://user-images.githubusercontent.com/2699324/62911697-e644aa00-bd4a-11e9-906d-c8384e3b6a5c.png)

After some internet sleuthing, I assumed I was running in the Byte Order Mark issue that you and @SukantGujar had stumbled across earlier and switched to the following npm run script

```javascript
"buildStatsWin": "webpack --profile --json | Out-file 'dist/stats.json' -Encoding OEM // windows
```

this ultimately failed as well with the following error

![image](https://user-images.githubusercontent.com/2699324/62911903-9adecb80-bd4b-11e9-8277-cbbfa280d745.png)

which is probably caused by some node to powershell piping issue but I'm not exactly certain. Running the second command from the command line seemed worked great and generated a stats.json file but I was still unable to run the webpack bundle analyzer.

As it turns out, the initial webpack command works perfectly and the powershell piping script works great but for some reason the piping in both scripts (I'm running these on a Windows 10 machine)  adds a new line to the stat.json with the path from which the the command was run. When I removed that line, everything started to work and bundle analyzer was up and running.

I'm still working on the upgraded branch of my code in [JustBlog](https://github.com/BrianMikinski/JustBlog) that will have on display but I"ll ping this thread if anyone wants to take a look.

Not sure where something obscure like this would belong in the documentation but I'm happy to help add it if you think other would benefit.

**Comment by valscion** [comment]
Hmm maybe something about npm scripts could warrant one sentence in the docs, maybe linking to your comment here evwn?
